shared-libs/transitions/src/transitions/index.js

```diff
@@ -0,0 +1,331 @@
+
+    const disabled = conf && conf.disable;
+    const allowSync = conf && !conf.async;
+
+    if (!conf || disabled || (!allowSync && synchronous)) {
+      return logger.warn(`Disabled transition "${transition}"`);
+    }
+
+    try {
+      transitions.push(self._loadTransition(transition));
+    } catch (e) {
+      loadError = true;
+      logger.error(`Failed loading transition "${transition}"`);
+      logger.error('%o', e);
+    }
+  });
+
+  // Warn if there are configured transitions that are not available
+  Object.keys(transitionsConfig).forEach(key => {
+    if (!AVAILABLE_TRANSITIONS.includes(key)) {
+      loadError = true;
+      logger.error(`Unknown transition "${key}"`);
+    }
+  });
+
+  if (loadError) {
+    throw new Error('Transitions are disabled until the above configuration errors are fixed.');
+  } else {
+    return transitions;
+  }
+};
+
+const loadTransition = key => {
+  logger.info(`Loading transition "${key}"`);
+  const transition = require('./' + key);
+  if (transition.init) {
+    transition.init();
+  }
+
+  return { key: key, module: transition };
+};
+
+/*
+ * Determines whether a transition is okay to run on a document.  Removed
+ * repeatable logic, assuming all transitions are repeatable without adverse
+ * effects.
+ */
+const canRun = ({ key, change, transition }) => {
+  const doc = change.doc;
+  const info = change.info;
+
+  const isRevSame = (doc, info) => {
+    if (info.transitions && info.transitions[key]) {
+      return parseInt(doc._rev) === parseInt(info.transitions[key].last_rev);
+    }
+    if (doc.transitions && doc.transitions[key]) {
+      return parseInt(doc._rev) === parseInt(doc.transitions[key].last_rev);
+    }
+    logger.debug(
+      `isRevSame tested true on transition ${key} for doc ${change.id} seq ${change.seq}`
+    );
+    return false;
+  };
+
+  /*
+   * Ignore deleted records, confirm transition has filter function and
+   * skip transition if filter returns false.
+   *
+   * If transition was already applied then only run again if doc rev is not
+   * equal to transition rev.  If revs are the same then transition just ran.
+   */
+  return Boolean(
+    change &&
+      !change.deleted &&
+      doc &&
+      !isRevSame(doc, info) &&
+      transition.filter(doc, info)
+  );
+};
+
+/*
+ * All transitions reference the same change.doc and work in series to
+ * apply changes to it.  A transition is free to make async calls but the next
+ * transition will only run after the previous transitions's callback is
+ * called.  This is a performance optimization that allows us to apply N
+ * transitions (updates) to a document with the cost of a single database
+ * change/write.
+ */
+const applyTransition = ({ key, change, transition }, callback) => {
+  if (!canRun({ key, change, transition })) {
+    logger.debug(
+      `canRun test failed on transition ${transition.key} for doc ${change.id} seq ${change.seq}`
+    );
+    return callback();
+  }
+
```

Comment:
I moved checking the call to `canRun` from the `applyTransitions` map function. 
The reason was to test `canRun` over the document as it has been updated so far. 

As an example, it enables the possibility of running `registration` transition over a fresh doc: currently registration requires either the form to be `public` or the `doc.contact` to exist, while `update_clinics` is responsible for creating `doc.contact` (registration runs `canRun` over an json already updated by `update_clinics`).